cloud/endagaweb/views/network.py

```diff
@@ -453,3 +455,193 @@ def get(self, request, network_id):
+            }
+            denom = models.NetworkDenomination.objects.get(id=dnm_id)
+            denom_data = {
+                'id': denom.id,
+                'start_amount': humanize_credits(denom.start_amount,
+                                                 CURRENCIES[currency]).amount,
+                'end_amount': humanize_credits(denom.end_amount,
+                                               CURRENCIES[currency]).amount,
+                'validity_days': denom.validity_days
+            }
+            response["data"] = denom_data
+            return http.HttpResponse(json.dumps(response),
+                                     content_type="application/json")
+
+        # Configure the table of denominations. Do not show any pagination
+        # controls if the total number of donominations is small.
+        if not user_profile.user.is_staff:
+            denom_table = django_tables.DenominationListTable(list(denom))
+        else:
+            denom_table = django_tables.DenominationTable(list(denom))
+        towers_per_page = 8
+        paginate = False
+        if denom > towers_per_page:
+            paginate = {'per_page': towers_per_page}
+        tables.RequestConfig(request, paginate=paginate).configure(denom_table)
+
+        # Set the context with various stats.
+        context = {
+            'networks': get_objects_for_user(request.user, 'view_network',
+                                             klass=models.Network),
+            'currency': CURRENCIES[user_profile.network.subscriber_currency],
+            'user_profile': user_profile,
+            'network': network,
+            'number_country': NUMBER_COUNTRIES[network.number_country],
+            'denomination': denom_count,
+            'denominations_table': denom_table,
+        }
+        # Render template.
+        info_template = template.loader.get_template(
+            'dashboard/network_detail/denomination.html')
+        html = info_template.render(context, request)
+        return http.HttpResponse(html)
+
+    def post(self, request):
+        """Operators can use this API to add denomination to a network.
+
+        These denomination bracket will be used to recharge subscriber, 
+        set balance validity and status
+        """
+        user_profile = models.UserProfile.objects.get(user=request.user)
+        network = user_profile.network
+        try:
+            currency = network.subscriber_currency
+            start_amount_raw = request.POST.get('start_amount')
+            start_amount = parse_credits(start_amount_raw,
+                                         CURRENCIES[currency]).amount_raw
+            end_amount_raw = request.POST.get('end_amount')
+            end_amount = parse_credits(end_amount_raw,
+                                       CURRENCIES[currency]).amount_raw
+            validity_days = int(request.POST.get('validity_days')) or 0
+            if validity_days > 10000:
+                validity_days = 10000
```

Comment:
NIT: Raise an error rather than forcing a maximum? Just saying you can do a lot of these checks in subclasses of `forms.Form` and raise `forms.ValidationError` and keep the post handler for more involved checks of combined fields/writing